packages/grid/_modules_/grid/hooks/features/filter/useGridFilter.ts

```diff
@@ -305,6 +310,33 @@ export const useGridFilter = (apiRef: GridApiRef, rowsProp: GridRowsProp): void
     }
   }, [apiRef, clearFilteredRows, logger, rowsProp]);
 
+  const prevFilterModelRef = React.useRef();
+
+  React.useEffect(() => {
+    return apiRef.current.subscribeEvent(GRID_FILTER_MODEL_CHANGE, (params) => {
+      const prevFilterModel = prevFilterModelRef.current;
+      prevFilterModelRef.current = params.filterModel;
+
+      const trimFilterModel = (filterModel) => ({
+        ...filterModel,
+        items: filterModel.items.filter((item) => item.value !== undefined),
```

Comment:
That should be handled by applyFilters. I don't think it should be in the effect